### ISSUE TITLE
Cannot read SFI table anymore

### ISSUE THREAD

**Issue by btashton** [andy-shev/u-boot#3]
Previous good output at boot:

(for reference this is the branch I built https://github.com/btashton/u-boot/tree/sfi_good)
```
U-Boot 2017.01-rc2-00042-gaa373aa (Jan 29 2017 - 15:56:13 -0800)
<SNIP>
Valid Boot Flag
Setup Size = 0x00003e00
Magic signature found
Using boot protocol version 2.0d
Linux kernel version 4.10.0-rc3 (bashton@localhost.localdomain) #1 SMP Fri Jan 20 21:57:24 PST 2017
Building boot_params at 0x00090000
Loading bzImage at address 100000 (6747200 bytes)
Magic signature found
Initial RAM disk at linear address 0x03000000, size 16777216 bytes
Kernel command line: "console=tty1 console=ttyS2,115200n8 rootfstype=ramfs rw"

Starting kernel ...
```

With your current edison branch
```
U-Boot 2017.01-00039-ga72dcec-dirty (Jan 29 2017 - 14:25:20 -0800)
<SNIP>
Valid Boot Flag
Setup Size = 0x00003e00 
Magic signature found   
Using boot protocol version 2.0d
Linux kernel version 4.10.0-rc3 (bashton@localhost.localdomain) #1 SMP Fri Jan 20 21:57:24 PST 2017
Building boot_params at 0x00090000
Loading bzImage at address 100000 (6747200 bytes)
Magic signature found   
ERROR: failed to locate SFI MMAP table

at arch/x86/cpu/tangier/sdram.c:119/sfi_setup_e820()
Initial RAM disk at linear address 0x03000000, size 16777216 bytes
Kernel command line: "console=tty1 console=ttyS2,115200n8 rootfstype=ramfs rw"

Starting kernel ...
```

This kernel then hangs until the watchdog and then when it resets it does not seem that u-boot is being loaded by ifwi

**Comment by btashton** [comment]
@andy-shev Should be resolved by this this #4 

I am wondering if there is any reason to grab only the lower 32 bits of pentry rather than all 64 as I have now.

**Comment by andy-shev** [comment]
Thanks for the fix!
It should be actually 64, but SFI is too 32-bit oriented, thus no difference.
Okay, based on your comment I'm closing this issue. Please, answer to my question there.
